htlcswitch/link.go

```diff
@@ -2352,13 +2396,10 @@ func (l *channelLink) CheckHtlcForward(payHash [32]byte,
 
 		// As part of the returned error, we'll send our latest routing
 		// policy so the sending node obtains the most up to date data.
-		failure := l.createFailureWithUpdate(
-			func(upd *lnwire.ChannelUpdate) lnwire.FailureMessage {
-				return lnwire.NewFeeInsufficient(
-					amtToForward, *upd,
-				)
-			},
-		)
+		cb := func(upd *lnwire.ChannelUpdate) lnwire.FailureMessage {
```

Comment:
Alternatively, couldn't we just masks this new logic _behind_ the existing `createFailureWithUpdate` method? As at this abstraction level, all that matters it that the link is able to obtain a channel update to send back.

Reply:
Yes createFailureWithUpdate could instead be changed, will put that on TODO

Reply:
done